### PR TITLE
Chore/Update app ids, add readme

### PR DESCRIPTION
**Pull request by friedger**
This PR adds the task `updateDemos` that verifies 
* that the application id follows the pattern `com.novoda.demo.<path>`
* that a readme exists for all demos
* that all demos are listed in the global readme file.
* ~~that all demos use the android-command plugin.~~

A change log file is also added.

### CODE REVIEW

**Comment by blundell** [comment]
:+1: 

 > that all demos use the android-command plugin.

what is the benefit of this? (we want to keep demo's simple so people can just copy pasta)

**Comment by friedger** [comment]
you can git clone and then just runDebug

**Comment by blundell** [comment]
I'm a bit confused tbh, demo's should be simple and generic - including our plugin in them all seems like overkill, nobody will know to do `runDebug` everyone just does `gradlew build`

**Comment by friedger** [comment]
We should teach developers to use it. That is what the demos are for.
 Am 26.06.2015 13:44 schrieb "Paul Blundell" <notifications@github.com>:

**Comment by blundell** [comment]
@friedger they're not :-( each demo should be demoing one individual feature, it shouldn't need a learning curve of our plugin :x:

**Comment by friedger** [comment]
it is optional to use

**Comment by friedger** [comment]
@blundell I have removed it never the less.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Change Log
+
+Changes for each demo are documented in the corresponding change log files.
+
+## June 2015
+Add android-command plugin. Run demos with
+```
+./gradlew ListStyles:runDebug
```
Comment (blundell) [inline CHANGELOG.md]: is this still valid?

File: build.gradle [hunk 2]
```diff
@@ -14,5 +15,51 @@ allprojects {
 }
 
 task wrapper(type: Wrapper) {
-    gradleVersion = '2.2.1'
-}
\ No newline at end of file
+    gradleVersion = '2.4'
+}
+
+task updateDemos() << {
```
Comment (blundell) [inline build.gradle]: can this be deleted now? cool script tho

Comment (friedger) [inline build.gradle]: I would leave the script or we can move it buildSrc. Running it from time
to time makes sure that everything is as we want it.
Am 26.06.2015 17:52 schrieb "Paul Blundell" <notifications@github.com>:

Comment (blundell) [inline build.gradle]: move it to /team-props/ or something out of the way then yeah :+1:

**Comment by blundell** [comment]
nice :+1:

**Pull request merged**